x/staking/keeper/grpc_query_test.go

```diff
@@ -0,0 +1,699 @@
+package keeper_test
+
+import (
+	gocontext "context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+	abci "github.com/tendermint/tendermint/abci/types"
+
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	"github.com/cosmos/cosmos-sdk/simapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/query"
+	"github.com/cosmos/cosmos-sdk/x/staking/keeper"
+	"github.com/cosmos/cosmos-sdk/x/staking/types"
+)
+
+type KeeperTestSuite struct {
+	suite.Suite
+
+	app         *simapp.SimApp
+	ctx         sdk.Context
+	addrs       []sdk.AccAddress
+	vals        []types.Validator
+	queryClient types.QueryClient
+}
+
+func (suite *KeeperTestSuite) SetupTest() {
+	app := simapp.Setup(false)
+	ctx := app.BaseApp.NewContext(false, abci.Header{})
+
+	querier := keeper.Querier{Keeper: app.StakingKeeper}
+
+	queryHelper := baseapp.NewQueryServerTestHelper(ctx, app.InterfaceRegistry())
+	types.RegisterQueryServer(queryHelper, querier)
+	queryClient := types.NewQueryClient(queryHelper)
+
+	addrs, _, validators := createValidators(ctx, app, []int64{9, 8, 7})
+	header := abci.Header{
+		ChainID: "HelloChain",
+		Height:  5,
+	}
+
+	hi := types.NewHistoricalInfo(header, validators)
+	app.StakingKeeper.SetHistoricalInfo(ctx, 5, hi)
+
+	suite.app, suite.ctx, suite.queryClient, suite.addrs, suite.vals = app, ctx, queryClient, addrs, validators
+}
+
+func (suite *KeeperTestSuite) TestGRPCQueryValidators() {
+	queryClient, vals := suite.queryClient, suite.vals
+	var (
+		req *types.QueryValidatorsRequest
+	)
+	testCases := []struct {
+		msg      string
+		malleate func()
+		expPass  bool
+	}{
+		{
+			"empty request",
+			func() {
+				req = &types.QueryValidatorsRequest{}
+			},
+			false,
+		},
+		{"valid request",
+			func() {
+				req = &types.QueryValidatorsRequest{Status: sdk.Bonded.String(),
+					Req: &query.PageRequest{Limit: 1, CountTotal: true}}
+			},
+			true,
+		},
```

Comment:
add cases for all the statuses

Reply:
I feel it will be redundant, as it works same way for all status. Do you think there's a need for those cases